pkg/datamover/repository_server_test.go

```diff
@@ -0,0 +1,199 @@
+	// Create Test Namespace
+	rss.namespace, err = createRepositoryServerTestNamespace(rss.ctx, rss.cli)
+	c.Assert(err, IsNil)
+
+	// Configure and Create Test S3 Credential and Location Secrets
+	rss.s3Creds, rss.s3Location, err = createTestS3CredsLocationSecrets(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Configure and Create Test User Access Secret for Kopia Repository Server
+	rss.userAccessSecret, err = createTestKopiaRepositoryServerUserAccessSecret(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Configure and Create Test TLS Certificate Secret
+	rss.tlsSecret, err = createTestKopiaTLSCertificateSecret(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Create Test Pod
+	rss.pod, err = createRepositoryServerTestPod(rss.ctx, rss.cli, rss.namespace.GetName(), rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Wait for Test Pod to get Ready
+	c.Assert(kube.WaitForPodReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod.Name), IsNil)
+
+	// Create Test Service
+	rss.service, err = createRepositoryServerTestService(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Configure and Create Kopia Repository
+	err = createTestKopiaRepository(rss.s3Location, rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Start Kopia Repository Server
+	err = startTestKopiaRepositoryServer(rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Wait for Kopia Repository Server To Get Ready
+	err = waitForServerReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Add Test Client User in Kopia Repository
+	err = addTestUserInKopiaRepositoryServer(rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Refresh Kopia Repository Server
+	err = refreshTestKopiaRepositoryServer(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Wait for Kopia Repository Server To Get Ready
+	err = waitForServerReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Set Kopia Repo Server Template Param
+	contentCacheMB, metadataCacheMB := kopiacmd.GetGeneralCacheSizeSettings()
+	rss.pod, err = rss.cli.CoreV1().Pods(rss.namespace.GetName()).Get(rss.ctx, rss.pod.GetName(), metav1.GetOptions{})
+	c.Assert(err, IsNil)
+	rss.repoServer = &param.RepositoryServer{
+		Name:      testRepoServerName,
+		Namespace: rss.namespace.GetName(),
+		ServerInfo: crv1alpha1.ServerInfo{
+			PodName:     rss.pod.GetName(),
+			ServiceName: rss.service.GetName(),
+		},
+		Username: testKopiaRepoServerUsername,
+		Credentials: param.RepositoryServerCredentials{
+			ServerTLS:        *rss.tlsSecret,
+			ServerUserAccess: *rss.userAccessSecret,
+		},
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+		Address:         fmt.Sprintf("https://%s:%d", rss.pod.Status.HostIP, rss.service.Spec.Ports[0].NodePort),
```

Comment:
Dont we want the server to be in ready state here before we execute kando location push? where are we setting `repositoryServer.Status.IsReady` to be true? or do we not check that in `kando location push` if the kopia repository server is ready?

Reply:
Since we're not creating RepositoryServer CR, We're checking if server is in Ready State https://github.com/kanisterio/kanister/pull/2168/files#diff-d0af91bee2ad274650bf1c95908597bd996518dcdbf1c35db766fa19ddb2c06bR108